pkg/chains/signing/kms/kms.go

```diff
@@ -38,7 +41,31 @@ type Signer struct {
 
 // NewSigner returns a configured Signer
 func NewSigner(ctx context.Context, cfg config.KMSSigner, logger *zap.SugaredLogger) (*Signer, error) {
-	k, err := kms.Get(ctx, cfg.KMSRef, crypto.SHA256)
+	kmsOpts := []signature.RPCOption{}
+	rpcAuth := options.RPCAuth{
+		Address: cfg.Auth.Address,
+		OIDC: options.RPCAuthOIDC{
+			Role: cfg.Auth.Role,
+			Path: cfg.Auth.Path,
+		},
+	}
+	// get token from spire socket
+	if cfg.Auth.Spire.Sock != "" {
```

Comment:
could all spire/kms related code be moved into a separate function for readability

```
function newSPIRESigner(ctx context.Context, cfg config.KMSSigner, logger *zap.SugaredLogger) (*Signer, error)
```

& we can return that `if cfg.Auth.Spire.Sock != ""`